src/vs/platform/telemetry/node/1dsAppender.ts

```diff
@@ -0,0 +1,155 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import type { IExtendedConfiguration, AppInsightsCore } from '@microsoft/1ds-core-js';
+import { IChannelConfiguration, IPayloadData, IXHROverride, PostChannel } from '@microsoft/1ds-post-js';
+import * as https from 'https';
+import { onUnexpectedError } from 'vs/base/common/errors';
+import { mixin } from 'vs/base/common/objects';
+import { ITelemetryAppender, validateTelemetryData } from 'vs/platform/telemetry/common/telemetryUtils';
+
+async function getClient(instrumentationKey: string): Promise<AppInsightsCore> {
+	const oneDs = await import('@microsoft/1ds-core-js');
+	const appInsightsCore = new oneDs.AppInsightsCore();
+	const collectorChannelPlugin: PostChannel = new PostChannel();
+	// Configure the app insights core to send to collector++ and disable logging of debug info
+	const coreConfig: IExtendedConfiguration = {
+		instrumentationKey,
+		endpointUrl: 'https://mobile.events.data.microsoft.com/collect/v1',
+		loggingLevelTelemetry: 0,
+		loggingLevelConsole: 0,
+		channels: [[
+			collectorChannelPlugin
+		]]
+	};
+
+	// Setup the collector posting channel to utilize nodes HTTP request rather than webs
+	if (coreConfig.extensionConfig) {
+		const customHttpXHROverride: IXHROverride = {
+			sendPOST: (payload: IPayloadData, oncomplete) => {
```

Comment:
You should wrap in a try / catch, as if you don't call the oncomplete() in ALL cases this can cause events to stop flowing.

Reply:
Actually, we handle this override throwing by logging a warning to the console and internally calling the oncomplete() when an exception is thrown... So this is not critical, but if you have any other mechanism for handling failures your might want to consider logging the error as well (and still calling oncomplete())